### PR TITLE
Makes Fireball Less Disgusting

### PR DESCRIPTION
**Pull request by GreatBigFailure**
Changes some things with fireball to make it less no-fun shuttlecall. Primarily a nerf but not entirely.

**Details**
1. Now costs 15 not 20

2. The explosion has been significantly reduced at base level. Impact damage slightly reduced.

3. It now requires robes to cast. I have no idea why it was no robes before.

4. Power upgrade that turns it targetable changed to "move" upgrade. Reduced its cost from 20 to 10. 

5. Power upgrade now has 3 levels, each cost 5. Each level increases explosive power. At level 3 it has the same explosion as current fireball. 

6. New "special" upgrade removes robe requirements, costs 10.

7. Cooldown increased to 20 seconds. Cooldown cost reduced(?) to 10. Might have been 10 already from a recent PR.

### **Update**
Buffed it slightly, creates a plume of fire when it explodes, base is a range of 1 around the explosion. Power level 1 increases it to 2.
The fire's damage scales a bit with power upgrades and a teeny tiny amount with every upgrade level. 

**tl;dr**
Base fireball is cheaper and weaker. Now costs 30 to get the same breaching power as current fireball and 40 to get the entire no robes function. Fireball does however now set things on fire in an AOE. 

:cl:
 * tweak: Fireball is now more flexible and less instant shuttle-call-y

### CODE REVIEW

**Comment by gurfan** [comment]
At what power level will this cause breaches. I think the main issue with this spell was the zero-effect breach ability rather than the killing power. The ability to breach easily was what got fuckup boots shitcanned, at least I think.

**Comment by GreatBigFailure** [comment]
> At what power level will this cause breaches

3

**Comment by DrSnips** [comment]
Why does the no robes exist as an upgrade? We have an existing entire no robes "spell" that people should have to buy instead

**Comment by GreatBigFailure** [comment]
> Why does the no robes exist as an upgrade? We have an existing entire no robes "spell" that people should have to buy instead

Because I didn't want to just gut the spell. It should be achievable in its current form, just not for 20 points. No Clothes also costs 40 while this upgrade costs 10.

**Comment by gurfan** [comment]
> Why does the no robes exist as an upgrade? We have an existing entire no robes "spell" that people should have to buy instead

I've never seen any wizard take no robes. 40 points is stupidly expensive for it. I'd rather give all robed spells an upgrade that removes the robe requirement.

**Comment by GreatBigFailure** [comment]
> > Why does the no robes exist as an upgrade? We have an existing entire no robes "spell" that people should have to buy instead
> 
> I've never seen any wizard take no robes. 40 points is stupidly expensive for it. I'd rather give all robed spells an upgrade that removes the robe requirement.

This isn't the only spell I want to tackle so I might add a few to others, ya. Robe requirements are a pretty interesting element of wizard that isn't touched on enough. Not in this PR though, unatomic and all that.

**Comment by ShiftyRail** [comment]
The point of fireball is ZAS meme. Please at least make it targettable by default then.

**Comment by GreatBigFailure** [comment]
Then upgrade it. Fireball is near universally hated as just ZAS hell shuttle call. 
I didn't make it targettable by default because there's a lot of advantages to not having it target. You can cast it with your hands full or without needing to aim your shot. A lot of people don't take target for the hand thing, might make it 5 cost instead for those reasons.

**Comment by ShiftyRail** [comment]
>Fireball is near universally hated as just ZAS hell shuttle call.

That's just baby talk. It doesn't kill anyone.

**Comment by GreatBigFailure** [comment]
Killing the crew isn't what's wrong with fireball. It's that you can cast a ranged C4 every 10 seconds without clothes on for base spell cost. It ends rounds immediately. There's a reason I didn't touch the damage too much.

**Comment by SonixApache** [comment]
fireball is great because it has a tendency to kill the wizard
losing this makes me sad

**Comment by githubuser4141** [comment]
How will the wizard cope with ZAS since he can't wear magboots anymore?

**Comment by GreatBigFailure** [comment]
> How will the wizard cope with ZAS since he can't wear magboots anymore?

Buy the upgrade, teleport out of it, buckle himself, wheelchair wizard, etc. Plenty of ways.

>fireball is great because it has a tendency to kill the wizard

It still can and will. Upgrading it makes it identical to current fireball. In fact the removal of no clothes makes it even more likely to kill them via ZAS if they aren't careful/don't upgrade. Base level probably won't kill the wizard though.

**Comment by GreatBigFailure** [comment]
>8:5

I'm surprised this is so mixed. I've never seen any positive opinions about fireball wizards, even those who don't do the full gemsuit setup. 
If you haven't posted your criticism of the changes then please let me know. I've been stubborn that fireball = bad, but I'm sure there's a compromise somewhere.

**Comment by AI-Is-Rogue** [comment]
Personally I would prefer breaching be an upgrade and that base fireball sets people on fire in an area. I just think that having the ability to telepathically control every firelock on the station is too much power even for a wizard.

**Comment by GreatBigFailure** [comment]
With this PR breaching is an upgrade. I didn't test it on all surfaces but on normal station tiles it won't breach until power rank 3, which costs 15 points total. As for setting people on fire I considered adding that but didn't want to change too much over-all in this PR. I'll try adding it and test it either tonight or tomorrow.

**Comment by Allspice** [comment]
Did you test the damage before/after, because some eyeballing suggest those base numbers are shite

**Comment by GreatBigFailure** [comment]
> Did you test the damage before/after, because some eyeballing suggest those base numbers are shite

Of course. The base is definitely a threat with a direct impact but really shit without one. That's mostly due to how explosions work, almost no room between an explosion that hurts and an explosion that breaches.
I'll add the fire thing that was suggested, that should help even out the difference.

**Comment by Allspice** [comment]
What is "a threat", because I see that "Impact damage slightly reduced" refers to changing the impact damage from 45 combined damage to 25.

**Comment by GreatBigFailure** [comment]
>impact damage from 45 combined damage to 25.

I actually meant to change it from 10:15 to 15:20, I must have forgotten. I'll change that when I PR the fire thing as well, good point. While testing a good chunk of the damage came from the explosion though, it would almost always break a few bones and cause internal bleeding.

**Comment by Allspice** [comment]
Other thing is that as I understand it, fireball doesn't actually "direct impact" people, as https://github.com/vgstation-coders/vgstation13/pull/12442 seems to suggest. I'm not sure that the reduced-range explosion even damages people outside of unusual circumstances.

**Comment by GreatBigFailure** [comment]
No it definitely does. I was worried about that too but if it hits you it definitely does the direct impact damage. I tinkered with the numbers a lot to make sure.

**Comment by GreatBigFailure** [comment]
Okay I updated it as promised. **Changes:**

- Now creates a plume of fire around the fireball when it explodes. At base it's 1 tile around it.
- Power level 1 increases it to 2 tiles around it.
- Every level of power increases fire damage moderately
- Every upgrade level in general increases it very, very slightly
- Reduced brute damage on upgrade slightly because all this stacked on top of each other was ridiculous

I also kept the accidentally lowered impact damage rather than changing it as with the fire it ended up being a much more reasonable amount of damage.

**Comment by Theophrastus1** [comment]
the only reason it's a "no fun shuttlecall" is because people refuse to rebuild rooms and fix the air. It's retarded

**Comment by GreatBigFailure** [comment]
Depends on the wizard round. Sometimes that's the case but it's very frequent to have ZAS just destroy everything by flinging items everywhere, kill a lot of people over time, just ruin the ability to move around the station properly. 
With the vote ratio I don't think this will be merged in its current state. Frankly I think I accidentally buffed fireball with the latest addition too. 

Gonna leave it up for another day or two in the hopes people give more feedback. I think current fireball absolutely needs to be changed but maybe I went too far.

**Comment by juanototo2** [comment]
If this gets merged, fball apprentices will be effectively unusable.

**Comment by Allspice** [comment]
@GreatBigFailure less controversial maybe if all you did was remove heavy radius and keep everything else the same
fucks up broom+fireball-from-space wizard, probably makes it better for wizards using it on station.

**Comment by GreatBigFailure** [comment]
>If this gets merged, fball apprentices will be effectively unusable.

That's a very good point. I'm going to rework this a bit again. This one suffers a bit from me trying to push how I think upgrades should work. I'm going to keep the explosion similar at base level, reduce impact damage, keep the fire plume I added to compensate (damage should be more spread in an AOE rather than single target to differentiate it), and make the breach a single level upgrade for 10-15? Keep the robeless upgrade.
Leaning toward keeping the cost at 15 then make power cost 15 so that it doesn't become a "might as well" upgrade for half the cost of base and instead doubles the cost but the end cost is the same.

So stronger at base, breach is limited by cost still. A bit more interesting than just being a rocket launcher due to fire plume.
Y/N?

**Comment by Blithering** [comment]
I'd say the targeting upgrade should be kept expensive, as it severely reduces the suicide factor.

**Comment by DamianX** [comment]
Votes are split even if people don't seem to hate it. 24 days ago you said you would rework this again but the last commit is from 26 days ago. Feel free to try again with a new PR.
